### ISSUE TITLE
Istioctl manifest generates hpa when not specified

### ISSUE THREAD

**Issue by harpratap** [istio/istio#25683]
**Bug description**
`istioctl manifest generate -f values.yaml` creates a default HPA for istio-ingressgateway deployment even though it wasn't specified. This creates an issue where if you define 20 replicas of gateway, the default hpa will kick in and reduce it to 5. We don't want to rely on HPA at all, hence prefer not to use it.

**Affected product area (please put an X in all that apply)**
[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Expected behavior**
HPA is defined only when specified (I think it's okay to add an HPA in default or demo profile, but doesn't make sense when my custom values.yaml also gets one autogenerated when I didn't specify it)

**Steps to reproduce the bug**
istioctl manifest generate -f values.yaml
```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    ingressGateways:
    - enabled: true
      k8s:
        replicaCount: 10
        resources:
          requests:
            cpu: 2000m
            memory: 512Mi
          limits:
            cpu: 2000m
            memory: 512Mi
        strategy:
          rollingUpdate:
            maxSurge: 100%
            maxUnavailable: 0%
        service:
          type: NodePort
          ports:
          - port: 80
            targetPort: 80
            name: http2
            nodePort: 60000
          - port: 443
            targetPort: 443
            name: https
            nodePort: 60001
      name: istio-ingressgateway
  profile: empty
  values:
    global:
      defaultPodDisruptionBudget:
        enabled: false
      jwtPolicy: first-party-jwt
```
**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
client version: 1.6.5
**How was Istio installed?**
istioctl manifest generate -f values.yaml | kubectl apply -f -

**Environment where bug was observed (cloud vendor, OS, etc)**
Locally on OSX and Linux

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by Quarky9** [comment]
We ran into the same "issue", but you can disable the hpa by setting:
```
values:
  gateways:
    istio-ingressgateway:
      autoscaleEnabled: false
```
via the Helm values.

**Comment by harpratap** [comment]
@Quarky9 Thanks a lot! 
Maybe it's better to move this to `kubernetesResourceSpec.hpaSpec.Enabled` 
I wasn't able to discover this setting in the documentation for helm values here either - https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/

**Comment by esnible** [comment]
@sdake @ostromart Perhaps Environments can work with Documentation to create a document based on the output of 1.7's `istioctl profile dump -o flags`.  Users have a hard time finding the options.

Using _master_ I did `istioctl profile dump -o flags | grep -i autoscale` which produces the output

```
values.gateways.istio-egressgateway.autoscaleEnabled=true
values.gateways.istio-ingressgateway.autoscaleEnabled=true
values.mixer.policy.autoscaleEnabled=true
values.mixer.telemetry.autoscaleEnabled=true
values.pilot.autoscaleEnabled=true
values.pilot.autoscaleMax=5
values.pilot.autoscaleMin=1
```

@ostromart Would it make sense if the default profiles specified <x>.autoscaleMax and Min for every component, instead of just Pilot?  I am mystified what the min and max scaling values for the ingress and egress would be if I turned on scaling.
